cmake/ccache.cmake

```diff
@@ -1,56 +1,53 @@
 
-if (NOT ENABLE_CCACHE)
+set (CCACHE_MINIMUM_VERSION 3.3)
+
+if (CCACHE_VERSION VERSION_LESS_EQUAL ${CCACHE_MINIMUM_VERSION})
+    message(${RECONFIGURE_MESSAGE_LEVEL} "Using ccache: no (found ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION}), the minimum required version is ${CCACHE_MINIMUM_VERSION}")
     return()
 endif()
 
-if (CCACHE_FOUND AND NOT COMPILER_MATCHES_CCACHE)
-   execute_process(COMMAND ${CCACHE_FOUND} "-V" OUTPUT_VARIABLE CCACHE_VERSION)
-   string(REGEX REPLACE "ccache version ([0-9\\.]+).*" "\\1" CCACHE_VERSION ${CCACHE_VERSION})
-
-   if (CCACHE_VERSION VERSION_GREATER "3.2.0" OR NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-      message(STATUS "Using ccache: ${CCACHE_FOUND} (version ${CCACHE_VERSION})")
-      set(LAUNCHER ${CCACHE_FOUND})
-
-      # debian (debhelpers) set SOURCE_DATE_EPOCH environment variable, that is
-      # filled from the debian/changelog or current time.
-      #
-      # - 4.0+ ccache always includes this environment variable into the hash
-      #   of the manifest, which do not allow to use previous cache,
-      # - 4.2+ ccache ignores SOURCE_DATE_EPOCH for every file w/o __DATE__/__TIME__
-      #
-      # Exclude SOURCE_DATE_EPOCH env for ccache versions between [4.0, 4.2).
-      if (CCACHE_VERSION VERSION_GREATER_EQUAL "4.0" AND CCACHE_VERSION VERSION_LESS "4.2")
-         message(STATUS "Ignore SOURCE_DATE_EPOCH for ccache")
-         set(LAUNCHER env -u SOURCE_DATE_EPOCH ${CCACHE_FOUND})
-      endif()
-
-      set (CMAKE_CXX_COMPILER_LAUNCHER ${LAUNCHER} ${CMAKE_CXX_COMPILER_LAUNCHER})
-      set (CMAKE_C_COMPILER_LAUNCHER ${LAUNCHER} ${CMAKE_C_COMPILER_LAUNCHER})
-   else ()
-       message(${RECONFIGURE_MESSAGE_LEVEL} "Using ccache: No. Found ${CCACHE_FOUND} (version ${CCACHE_VERSION}) but disabled because of bug: https://bugzilla.samba.org/show_bug.cgi?id=8118")
-   endif ()
-elseif (NOT CCACHE_FOUND AND NOT COMPILER_MATCHES_CCACHE)
-    message (${RECONFIGURE_MESSAGE_LEVEL} "Using ccache: No")
-endif ()
+message(STATUS "Using ccache: ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION})")
+set(LAUNCHER ${CCACHE_EXECUTABLE})
+
+# Work around a well-intended but unfortunate behavior of ccache 4.0 & 4.1 with
+# environment variable SOURCE_DATE_EPOCH. This variable provides an alternative
+# to source-code embedded timestamps (__DATE__/__TIME__) and therefore helps with
+# reproducible builds (*). SOURCE_DATE_EPOCH is set automatically by the
+# distribution, e.g. Debian. Ccache 4.0 & 4.1 incorporate SOURCE_DATE_EPOCH into
+# the hash calculation regardless they contain timestamps or not. This invalidates
+# the cache whenever SOURCE_DATE_EPOCH changes. As a fix, ignore SOURCE_DATE_EPOCH.
+#
+# (*) https://reproducible-builds.org/specs/source-date-epoch/
+if (CCACHE_VERSION VERSION_GREATER_EQUAL "4.0" AND CCACHE_VERSION VERSION_LESS "4.2")
+    message(STATUS "Ignore SOURCE_DATE_EPOCH for ccache 4.1 / 4.2")
```

Comment:
```suggestion
    message(STATUS "Ignore SOURCE_DATE_EPOCH for ccache 4.0 / 4.1")
```

Reply:
Thanks (guess I need to get another coffee  :D).